linera-service/src/proxy/grpc.rs

```diff
@@ -491,17 +491,19 @@ where
 
         let mut certificates = vec![];
 
-        for certificate in self
-            .0
-            .storage
-            .read_certificates(hashes)
-            .await
-            .map_err(|err| Status::from_error(Box::new(err)))?
-        {
-            if grpc_message_limiter.fits::<Certificate>(certificate.clone())? {
-                certificates.push(certificate);
-            } else {
-                break;
+        for batch in hashes.chunks(100) {
+            for certificate in self
+                .0
+                .storage
+                .read_certificates(batch.to_vec())
+                .await
+                .map_err(|err| Status::from_error(Box::new(err)))?
+            {
+                if grpc_message_limiter.fits::<Certificate>(certificate.clone())? {
+                    certificates.push(certificate);
+                } else {
+                    break;
```

Comment:
We should break out of the _outer_ loop here.

Reply:
Ah, good point.

Reply:
Done (pushed with force).